homeassistant/components/roomba/sensor.py

```diff
@@ -114,143 +114,41 @@ def native_value(self):
         return self._run_stats.get("hr")
 
 
-class TotalMissions(IRobotEntity, SensorEntity):
-    """Class to hold Roomba Sensor basic info."""
-
-    _attr_entity_category = EntityCategory.DIAGNOSTIC
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return f"{self._name} missions total"
-
-    @property
-    def unique_id(self):
-        """Return the ID of this sensor."""
-        return f"total_missions_{self._blid}"
-
-    @property
-    def icon(self):
-        """Return the counter icon."""
-        return "mdi:counter"
-
-    @property
-    def native_unit_of_measurement(self):
-        """Return the unit_of_measurement of the device."""
-        return ""
-
-    @property
-    def native_value(self):
-        """Return the state of the sensor."""
-        return self._mission_stats.get("nMssn")
-
-
-class SuccessMissions(IRobotEntity, SensorEntity):
-    """Class to hold Roomba Sensor basic info."""
-
-    _attr_entity_category = EntityCategory.DIAGNOSTIC
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return f"{self._name} missions successful"
-
-    @property
-    def unique_id(self):
-        """Return the ID of this sensor."""
-        return f"successful_missions_{self._blid}"
-
-    @property
-    def native_unit_of_measurement(self):
-        """Return the unit_of_measurement of the device."""
-        return ""
-
-    @property
-    def icon(self):
-        """Return the counter icon."""
-        return "mdi:counter"
+class MissionSensor(IRobotEntity, SensorEntity):
+    """Class to hold the Roomba missions info."""
 
-    @property
-    def entity_registry_enabled_default(self):
-        """Disable sensor by default."""
-        return False
-
-    @property
-    def native_value(self):
-        """Return the state of the sensor."""
-        return self._mission_stats.get("nMssnOk")
-
-
-class CancelledMissions(IRobotEntity, SensorEntity):
-    """Class to hold Roomba Sensor basic info."""
+    def __init__(self, roomba, blid, mission_type, mission_value_string):
+        """Initialise iRobot sensor with mission details."""
+        super().__init__(roomba, blid)
+        self._mission_type = mission_type
+        self._mission_value_string = mission_value_string
 
     _attr_entity_category = EntityCategory.DIAGNOSTIC
 
     @property
     def name(self):
         """Return the name of the sensor."""
-        return f"{self._name} missions canceled"
+        return f"{self._name} missions {self._mission_type}"
 
     @property
     def unique_id(self):
         """Return the ID of this sensor."""
-        return f"canceled_missions_{self._blid}"
-
-    @property
-    def native_unit_of_measurement(self):
-        """Return the unit_of_measurement of the device."""
-        return ""
+        return f"total_missions_{self._blid}"
```

Comment:
```suggestion
        return f"{self._mission_type}_missions_{self._blid}"
```

You forgot to include the the mission type in the unique id. Apart from that, everything appears to be working.

Thank you very much for helping me here 😄